pkg/controller/core/workload_controller.go

```diff
@@ -194,16 +187,39 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if updated {
 			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
 		}
-	} else if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
-		// if job is not active and does not have condition reason of WorkloadEvictedByDeactivation, update its condition
-		workload.SetEvictedCondition(&wl, kueue.WorkloadEvictedByDeactivation, "The workload is deactivated")
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
-			return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
+	} else {
+		var updated, evicted bool
+		if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
+			message := "The workload is deactivated"
+
+			// The deactivation reason could be deduced as the maximum number of re-queuing retries if the workload met all criteria below:
+			//  1. The waitForPodsReady feature is enabled, which means that it has "PodsReady" condition.
+			//  2. The workload has already exceeded the PodsReadyTimeout.
+			//  3. The workload already has been re-queued previously, which means it doesn't have the requeueAt field.
+			//  4. The number of re-queued has already reached the waitForPodsReady.requeuingBackoffLimitCount.
+			if apimeta.IsStatusConditionFalse(wl.Status.Conditions, kueue.WorkloadPodsReady) &&
+				((!workload.HasRequeueState(&wl) && ptr.Equal(r.waitForPodsReady.requeuingBackoffLimitCount, ptr.To[int32](0))) ||
+					(workload.HasRequeueState(&wl) && wl.Status.RequeueState.RequeueAt == nil &&
+						ptr.Equal(wl.Status.RequeueState.Count, r.waitForPodsReady.requeuingBackoffLimitCount))) {
+				message = fmt.Sprintf("%s by exceeded the maximum number of re-queuing retries", message)
```

Comment:
```suggestion
				message = fmt.Sprintf("%s due to exceeding the maximum number of re-queuing retries", message)
```
nit, I think it will read better

Reply:
SGTM, thank you for this recommendation!

Reply:
Done.